docs/source/fetching/defer.mdx

```diff
@@ -30,18 +30,18 @@ query PersonQuery($personId: ID!) {
 }
 ```
 
-In the generated code for this query, the `asUser` field for the fragment will be optional. That is because when the initial payload is received from the server, the fields of the fragment are not yet present. A `Person` will be returned with only the basic fields filled in.
+In the generated code for this query the `asUser` type case will be optional, it will also have a `fragments` container in which there will be an optional fragment with the same name as the `label` property value used in the query. The deferred fragment is optional because it will not be returned with the basic fields, it will be returned in a separate response instead.
```

Comment:
Will there `asUser` even be generated? I'm not sure if the defer work changes this, but in the past, if the `person` selection set was already on `User`, a type case on the exact matching type would be reduced out by the codegen engine. I think the preferred behavior here would be to just have `data.person.fragments.deferredFriends`.

If this only changed for deferred type cases, I'm okay with it, but if this changes how the codegen engine handles all type cases, then we're going to have existing code break when regenerated.

Reply:
If person was on some `PersonInterface` type and the typecase was on `User`, then this would be correct though.

Reply:
There isn't any generated code for this example that I was working from, I simply copied the [example from Apollo Kotlin](https://www.apollographql.com/docs/kotlin/fetching/defer/) so the docs are similar.

> If person was on some PersonInterface type and the typecase was on User, then this would be correct though.

That's my understanding of how this generated model would work. `person` is defined as an interface and `friends` only exists on `User` which implements `PersonInterface`. I agree that if `person` was of type `User` then `asUser` would not be generated; defer hasn't changed that.